File: e2e/selenide-junit5-selenium-grid/src/test/java/data/TaskData.java
package data;

import lombok.Builder;
import lombok.Data;

import static org.apache.commons.lang3.RandomStringUtils.secure;

@Builder
@Data
public class TaskData {
    @Builder.Default
    private String title = secure().nextAlphabetic(12);

    @Builder.Default
    private String description = secure().nextAlphabetic(12);

    @Builder.Default
    private TaskStatus status = TaskStatus.TODO;

    @Builder.Default
    private TaskPriority priority = TaskPriority.MEDIUM;
}

File: demo-ios/DemoUITests/Data/TaskData.swift
struct TaskData: Equatable {
    let title: String
    let description: String
    let status: TaskStatus
    let priority: TaskPriority

    var allureLabel: String {
        "{title: \(title), description: \(description), status: \(status.rawValue), priority: \(priority.rawValue)}"
    }
}
